x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheMetrics.java

```diff
@@ -7,15 +7,36 @@
 
 package org.elasticsearch.blobcache;
 
+import org.elasticsearch.index.shard.ShardId;
 import org.elasticsearch.telemetry.TelemetryProvider;
 import org.elasticsearch.telemetry.metric.LongCounter;
 import org.elasticsearch.telemetry.metric.LongHistogram;
 import org.elasticsearch.telemetry.metric.MeterRegistry;
 
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
 public class BlobCacheMetrics {
+    private static final String CACHE_POPULATION_TYPE_ATTRIBUTE_KEY = "cachePopulationType";
+    private static final String SHARD_ID_ATTRIBUTE_KEY = "shardId";
+
     private final LongCounter cacheMissCounter;
     private final LongCounter evictedCountNonZeroFrequency;
     private final LongHistogram cacheMissLoadTimes;
+    private final LongHistogram cachePopulateReadThroughput;
+    private final LongHistogram cachePopulateWriteThroughput;
+    private final LongHistogram cachePopulationElapsedTime;
+
+    public enum CachePopulationReason {
+        /**
+         * When warming the cache
+         */
+        Warming,
+        /**
+         * When fetching a new commit
+         */
+        LoadCommit
+    }
```

Comment:
There's probably more to add here, e.g. cache-miss, but this is just based on the places I implemented in the serverless change built on top of this stuff so far. Perhaps it doesn't need to be an enum?

Reply:
Just looking at the cache-miss instrumentation, at this level of abstraction it's difficult/impossible to know if you're loading from the blob store or an indexing node. At least with `CachePopulationReason` we can exclude 'CacheMiss' if we only want metrics from blob store loads? I will keep thinking about this.